src/Workspaces/Core/Portable/Shared/Extensions/ProjectExtensions.cs

```diff
@@ -53,5 +54,27 @@ internal static Project WithSolutionOptions(this Project project, OptionSet opti
 
         private static DocumentId? GetDocumentIdWithFilePath(this Project project, string filePath)
             => project.Solution.GetDocumentIdsWithFilePath(filePath).FirstOrDefault(id => id.ProjectId == project.Id);
+
+        public static Document GetRequiredDocument(this Project project, DocumentId documentId)
```

Comment:
we already have a get required document, any reason we need to re-define here?
http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/ISolutionExtensions.cs,d56b2beb86c3b187

Reply:
I moved it to ProjectExtensions since it is a Project extension method. Would you rather have it stay in ISolutionExtensions?